www/_blog/2021-03-31-supabase-cli.mdx

```diff
@@ -0,0 +1,223 @@
+
+There was a problem in using the [pgAdmin Schema Diff](https://www.pgadmin.org/docs/pgadmin4/development/schema_diff.html) though, it's a GUI-only tool.
+
+![pgadmin diff](/new/images/blog/cli/pgadmin-diff.png)
+
+So we did what we always strive to do - improve existing open source software. We created a CLI mode for the Schema Diff on [our repo](https://github.com/supabase/pgadmin4/blob/cli/web/cli.py). We've also released a [docker image](https://hub.docker.com/r/supabase/pgadmin-schema-diff) for a quick start.
+
+The CLI offers the same functionality as the GUI version. You can diff two databases by specifying the connection strings like shown below.
+
+```bash
+docker run supabase/pgadmin-schema-diff \
+  'postgres://user:pass@local:5432/diff_source' \
+  'postgres://user:pass@production:5432/diff_target' \
+  > diff_demo.sql
+
+Starting schema diff...
+Comparision started......0%
+Comparing Event Triggers...2%
+Comparing Extensions...4%
+Comparing Languages...8%
+Comparing Foreign Servers...14%
+Comparing Foreign Tables of schema 'public'...28%
+Comparing Tables of schema 'public'...50%
+Comparing Domains of schema 'test_schema_diff'...66%
+Comparing Foreign Tables of schema 'test_schema_diff'...68%
+Comparing FTS Templates of schema 'test_schema_diff'...76%
+Comparing Functions of schema 'test_schema_diff'...78%
+Comparing Procedures of schema 'test_schema_diff'...80%
+Comparing Tables of schema 'test_schema_diff'...90%
+Comparing Types of schema 'test_schema_diff'...92%
+Comparing Materialized Views of schema 'test_schema_diff'...96%
+Done.
+
+## the diff is written to diff_demo.sql
+```
+
+A sample diff can be seen on this [gist](https://gist.github.com/steve-chavez/3f286a233806aeee0bcea4a47f97f0b5). This was generated by diffing these [two databases](https://github.com/supabase/pgadmin4/tree/cli/web/pgadmin/tools/schema_diff/tests/pg/10_plus).
+
+On these [lines](https://gist.github.com/steve-chavez/3f286a233806aeee0bcea4a47f97f0b5#file-diff_demo-sql-L1022-L1023), you can see how it tracks the view's owner change (note: the `ALTER TABLE` statement is interchangeable with `ALTER VIEW` in this case). Additionally, you can see that it handles [domains](https://gist.github.com/steve-chavez/3f286a233806aeee0bcea4a47f97f0b5#file-diff_demo-sql-L278-L287) just fine(this is an edge-case that other diff tools don't handle).
+
+Also, similarly to the pgAdmin GUI:
+
+- You can include and exclude database objects from the diff with `--include-objects` or `--exclude-objects`
+- You can choose a single schema to diff with the `--schema` argument. We recommend you do this for Supabase databases. Diffing the whole database can take a while because of the `extensions` schema (especially if you enable PostGIS, which adds many functions).
```

Comment:
```suggestion
- You can choose a single schema to diff with the `--schema` argument or you can pick different schemas to compare with the --source-schema and --target-schema arguments. We recommend you do this for Supabase databases. Diffing the whole database can take a while because of the `extensions` schema (especially if you enable PostGIS, which adds many functions).
```